### PR TITLE
update to modern addon style + bumps framework 7 version

### PR DESCRIPTION
**Pull request by jamesdixon**
This PR brings the add-on inline with current conventions for Ember add-ons. 

It also updates the Framework 7 dependency to the latest (~1.5.2).

### CODE REVIEW

File: tests/dummy/app/initializers/application.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import Ember from 'ember';
 
-export function initialize(container, application) {
+export function initialize(application) {
   Ember.run.schedule('afterRender', function() {
     container.lookup('service:framework7').initSwipePanels('left');
```
Comment (spieker) [inline tests/dummy/app/initializers/application.js]: This will not work, since there is no reference of `container` anymore.

File: README.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ need to initialize the listeners within an initializer.
 Framework7Initializer =
   name: 'framework7'
   after: 'framework7-service'
-  initialize: (container, application) ->
+  initialize: (application) ->
     Ember.run.schedule 'afterRender', ->
       container.lookup('service:framework7').initSwipePanels 'left'
```
Comment (spieker) [inline README.md]: `container` will be missing, right?

**Comment by jamesdixon** [comment]
@spieker no worries at all! Thanks for your initial work on this.

You're absolutely right. I'll look into correcting this as soon as I have some time.

Cheers,
James

**Comment by jamesdixon** [comment]
@spieker I just took a look at this and it's slightly more complicated than I thought. I think the entire dummy app needs to be looked into because there are still Ember Views 😄

**Comment by spieker** [comment]
@jamesdixon Uh, yes, that's true. Do you find time to deal with it?

**Comment by jamesdixon** [comment]
@spieker no time right now. In the midst of launching a product :)

**Comment by jamesdixon** [comment]
@spieker would you be open to merging these PRs and I'll work on fixing the dummy app later? I have a bunch of other changes/additions I'd like to make, but honestly, they're priority over the dummy app :)

**Pull request merged**